pumpkin-lib/src/engine/cp/assignments_integer.rs

```diff
@@ -348,9 +365,27 @@ impl AssignmentsInteger {
                 let value_before = self.domains[domain_id].lower_bound;
                 self.domains[domain_id].undo_trail_entry(&entry);
                 if fixed_before && self.domains[domain_id].lower_bound != self.domains[domain_id].upper_bound {
+                    // This `domain_id` was unassigned while backtracking
+                    self.backtrack_events.event_occurred(IntDomainEvent::Assign, domain_id);
```

Comment:
Here we should probably check if there are any watchers, otherwise the events are always pushed and never drained. Since backtracking is on the hot path, it is worth testing I think

Reply:
I have added this check by passing the `is_watching_any_backtrack_events` which we are tracking